var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -189,18 +195,17 @@ class Llvm(CMakePackage, CudaPackage):
     # https://bugs.llvm.org/show_bug.cgi?id=39696
     patch("thread-p9.patch", when="@develop+libcxx")
 
-    @run_before("cmake")
-    def check_darwin_lldb_codesign_requirement(self):
-        if not self.spec.satisfies("+lldb platform=darwin"):
-            return
-        codesign = which("codesign")
-        mkdir("tmp")
-        llvm_check_file = join_path("tmp", "llvm_check")
-        copy("/usr/bin/false", llvm_check_file)
-
+    @run_before('cmake')
+    if '+code_signing' in self.spec:
+        def codesign_check(self):
```

Comment:
Switch these two lines. Even if you don't have access to macOS, you can still test this by running `spack info llvm`. The current code definitely doesn't work.